trieste/acquisition/rule.py

```diff
@@ -148,6 +162,36 @@ def acquire_single(
             datasets=None if dataset is None else {OBJECTIVE: dataset},
         )
 
+    def update_datasets(
```

Comment:
Do we have to have this as a method of `AcquisitionRule`? 

In general an acquisition rule should not update any dataset, it should just receive a dataset and model and make a query point recommendation based on this. So this could confuse users.

Reply:
The reason is that the behaviour of datasets-update changes depending on which rule we have. For most rules the datasets are simply concatenated; but for `BatchTrustRegion` rule there is quite a bit of code to filter-out points outside the subspaces; and perhaps we will have other ways in the future. So it seemed sensible to encapsulate that functionality as part of the rule.

We could add it in the `BayesianOptimizer`/`AskTellOptimizer` instead (which is where the previous dataset update was). However, that code would then likely need to use dynamic type detection, and have different behaviour depending on the rule. That seemed more messy. We can certainly go down that route though.

Reply:
I have changed this so that dataset updates are back in `BayesianOptimizer`/`AskTellOptimizer`. The rule now contains only filtering of the datasets.

The reason for this change (other than this discussion) was that the previous version threw away points as soon as they were outside a region. If a region ever came back over the same space, those points were lost. The new version keeps the full datasets around. Hence a region can use points irrespective of when they were collected.